cmd/gcp-routes-controller/run.go

```diff
@@ -196,3 +196,152 @@ func (sl *healthTracker) OnComplete(state *health.State) {
 		}
 	}
 }
+
+func getRoutes() (map[string][]string, error) {
+	routes := make(map[string][]string)
+	const netPath = "network-interfaces/"
+	ifs, err := getInstances(netPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get instances at %s: %v", netPath, err)
+	}
+	for _, vif := range ifs {
+		macs, err := getInstances(netPath + vif + "mac")
+		if err != nil {
+			return nil, fmt.Errorf("failed to get instances at %s: %v", netPath+vif+"mac", err)
+		}
+		hwAddr := macs[0]
+		fwipPath := netPath + vif + "forwarded-ips/"
+		devName, err := getIfname(hwAddr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get interface name for %s: %v", hwAddr, err)
+		}
+		levels, err := getInstances(fwipPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get instances at %s: %v", fwipPath, err)
+		}
+		for _, level := range levels {
+			fwips, err := getInstances(fwipPath + level)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get instances at %s: %v", fwipPath+level, err)
+			}
+			glog.Info("Processing route for NIC " + vif + hwAddr + " as " + devName + " for IPs " + strings.Join(fwips, ", "))
+			routes[devName] = fwips
+		}
+	}
+
+	return routes, nil
+}
+
+func getIfname(hwAddr string) (string, error) {
+	devs, err := net.Interfaces()
+	if err != nil {
+		return "", fmt.Errorf("failed to list interfaces: %v", err)
+	}
+	for _, dev := range devs {
+		if dev.HardwareAddr.String() == hwAddr {
+			return dev.Name, nil
+		}
+	}
+
+	return "", fmt.Errorf("interface not found")
+}
+
+func getInstances(instanceType string) ([]string, error) {
+	const baseURL = "http://metadata.google.internal/computeMetadata/v1/instance/"
```

Comment:
How is this resolved? If it's not getting it from /etc/resolv.conf, I see the google libraries, by default, query 169.254.169.254 instead of depending on name resolution: https://github.com/googleapis/google-cloud-go/blob/eed250f8ad01692794e90c5862a7be21b1de2c51/compute/metadata/metadata.go

I'm not quite sure if an scneario where resolution doesn't work due to coreDNS or someting, but I don't see downsides of setting the ip url to use 169.254.169.254 rather depeding on name resolution.